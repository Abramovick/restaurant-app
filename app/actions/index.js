import * as types from '../constants/Actiontypes';

const receiveProducts = products => ({
  type: types.RECEIVE_PRODUCTS,
  products
});

const addToCartFn = product => ({
  type: types.ADD_TO_CART,
  product
});

const addToken = token => ({
  type: types.ADD_TOKEN,
  token
});

const login = loginState => ({
  type: types.LOGIN_FAIL,
  loginState
});

const checkOut = () => ({
  type: types.CHECKOUT,
});

const increaseQuantity = product => ({
  type: types.INCREASE_ITEM_QUANTITY,
  product
});

const decreaseQuantity = product => ({
  type: types.DECREASE_ITEM_QUANTITY,
  product
});

const clearAuth = () => ({
  type: types.CLEAR_AUTH,
});

export const addProducts = products => (dispatch) => {
  dispatch(receiveProducts(products));
};
export const addTokenToStore = token => (dispatch) => {
  dispatch(addToken(token));
};
export const addToCart = product => (dispatch) => {
  dispatch(addToCartFn(product));
};

export const increaseItemQuantity = product => (dispatch) => {
  product.quantity += 1;
  dispatch(increaseQuantity(product));
};

export const decreaseItemQuantity = product => (dispatch) => {
  product.quantity -= 1;
  dispatch(decreaseQuantity(product));
};

export const clearCart = () => (dispatch) => {
  dispatch(checkOut());
};

export const removeAuth = () => (dispatch) => {
  dispatch(clearAuth());
};

export const loginFail = loginState => (dispatch) => {
  dispatch(login(loginState));
};
