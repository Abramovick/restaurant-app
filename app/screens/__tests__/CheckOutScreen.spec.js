import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow, } from 'enzyme';
import ConnectedCheckOut, { CheckOutScreen } from '../CheckOutScreen';
import { clearCart } from '../../actions';

// Test objects
const data = {
  loggedInUser: {
    id: 345656
  }
};

const dataNoAuth = {
  loggedInUser: {
    id: ''
  }
};

const items = [{
  name: 'mariam',
  ImageUrl: 'gdfgds',
  Price: 700,
  description: 'the hello is in the world '
},
{
  name: 'tope',
  ImageUrl: 'gdfgds',
  Price: 700,
  description: 'the hello is in the world'
}];

const navigation = {
  navigate: jest.fn(),
  state: {
    params: {
      name: 'mariam',
      ImageUrl: 'gdfgds',
      Price: 700,
      description: 'the hello is in the world '
    }
  }
};

// Mocked functions
const checkOutMutation = jest.fn();
const closeModal = jest.fn();

// Test suites
describe('CHECKOUTSCREEN --- Snapshot', () => {
  it('renders the checkout screen if user is logged in', async () => {
    const tree = renderer.create(<CheckOutScreen
      items={items}
      navigation={navigation}
      clearCart={clearCart}
      checkOutMutation={checkOutMutation}
      data={data}
      closeModal={closeModal}
      />);
    expect(tree.getInstance()._isLoggedIn()).toMatchSnapshot();
    expect(tree.getInstance().completed()).toMatchSnapshot();
    expect(CheckOutScreen.navigationOptions(navigation)).toMatchSnapshot();
    expect(tree).toMatchSnapshot();
  });

  it('renders a logout/login view is user is not logged in', async () => {
    const tree = renderer.create(<CheckOutScreen
      items={items}
      navigation={navigation}
      clearCart={clearCart}
      data={dataNoAuth}
      checkOutMutation={checkOutMutation}
      closeModal={closeModal}
      />);
    expect(tree.getInstance()._isLoggedIn()).toMatchSnapshot();
    expect(tree.getInstance().completed()).toMatchSnapshot();
    expect(tree).toMatchSnapshot();
  });
});

describe('CHECKOUTSCREEN --- Shallow rendering + passing the store directly', () => {
  const initialState = {
    cart: {
      items
    },
    total: 453543
  };
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  let store, container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<ConnectedCheckOut store={store}/>);
  });

  it('renders the connected component', () => {
    expect(container.length).toEqual(1);
  });

  it('check Prop matches with initialState', () => {
    expect(container.prop('items'))
      .toEqual(initialState.cart.items);
    expect(container.prop('total'))
      .toEqual(initialState.cart.total);
  });

  it('checks action on dispatching ', () => {
    let action = '';
    store.dispatch(clearCart());
    action = store.getActions();
    expect(action[0].type).toBe('CHECKOUT');
  });
});
