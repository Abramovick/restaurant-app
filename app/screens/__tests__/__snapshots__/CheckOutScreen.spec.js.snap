// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CHECKOUTSCREEN --- Snapshot renders a logout/login view is user is not logged in 1`] = `false`;

exports[`CHECKOUTSCREEN --- Snapshot renders a logout/login view is user is not logged in 2`] = `undefined`;

exports[`CHECKOUTSCREEN --- Snapshot renders a logout/login view is user is not logged in 3`] = `undefined`;

exports[`CHECKOUTSCREEN --- Snapshot renders a logout/login view is user is not logged in 4`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Log out and Log in again
  </Text>
</View>
`;

exports[`CHECKOUTSCREEN --- Snapshot renders the checkout screen if user is logged in 1`] = `true`;

exports[`CHECKOUTSCREEN --- Snapshot renders the checkout screen if user is logged in 2`] = `undefined`;

exports[`CHECKOUTSCREEN --- Snapshot renders the checkout screen if user is logged in 3`] = `undefined`;

exports[`CHECKOUTSCREEN --- Snapshot renders the checkout screen if user is logged in 4`] = `
Object {
  "title": "Order Detail",
}
`;

exports[`CHECKOUTSCREEN --- Snapshot renders the checkout screen if user is logged in 5`] = `
<RCTScrollView>
  <View>
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutoAutomaticScroll={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      onScroll={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={0}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      virtual={undefined}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flex": 1,
                "flexWrap": "wrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "padding": 20,
              },
            ]
          }
          virtual={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              fontSize={15}
              style={
                Array [
                  Object {
                    "fontFamily": "SinkinSans-200XLight",
                    "fontSize": 12,
                  },
                  Object {
                    "fontSize": 15,
                  },
                ]
              }
            >
               ORDER PROCESSED
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "borderBottomColor": "#B2BEB5",
                  "borderBottomWidth": 0.5,
                  "height": 1,
                  "marginBottom": 20,
                  "marginTop": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              fontSize={15}
              style={
                Array [
                  Object {
                    "fontFamily": "SinkinSans-200XLight",
                    "fontSize": 12,
                  },
                  Object {
                    "fontSize": 15,
                  },
                ]
              }
            >
               PAYMENT METHOD :  PAY ON DELIVERY
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "borderBottomColor": "#B2BEB5",
                  "borderBottomWidth": 0.5,
                  "height": 1,
                  "marginBottom": 20,
                  "marginTop": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              fontSize={15}
              style={
                Array [
                  Object {
                    "fontFamily": "SinkinSans-200XLight",
                    "fontSize": 12,
                  },
                  Object {
                    "fontSize": 15,
                  },
                ]
              }
            >
              mariam
              :      
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              fontSize={15}
              style={
                Array [
                  Object {
                    "fontFamily": "SinkinSans-200XLight",
                    "fontSize": 12,
                  },
                  Object {
                    "fontSize": 15,
                  },
                ]
              }
            >
              tope
              :      
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "borderBottomColor": "#B2BEB5",
                  "borderBottomWidth": 0.5,
                  "height": 1,
                  "marginBottom": 20,
                  "marginTop": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              fontSize={15}
              style={
                Array [
                  Object {
                    "fontFamily": "SinkinSans-200XLight",
                    "fontSize": 12,
                  },
                  Object {
                    "fontSize": 15,
                  },
                ]
              }
            >
               TOTAL AMOUNT : ₦
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "borderBottomColor": "#B2BEB5",
                  "borderBottomWidth": 0.5,
                  "height": 1,
                  "marginBottom": 20,
                  "marginTop": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              fontSize={15}
              style={
                Array [
                  Object {
                    "fontFamily": "SinkinSans-200XLight",
                    "fontSize": 12,
                  },
                  Object {
                    "fontSize": 15,
                  },
                ]
              }
            >
               ENTER DELIVERY ADDRESS
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              placeholder="Enter the delivery address"
              style={
                Object {
                  "borderColor": "gray",
                  "borderWidth": 1,
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 10,
                  "height": 50,
                  "marginTop": 10,
                  "padding": 0,
                  "textAlignVertical": "top",
                  "width": "100%",
                }
              }
              underlineColorAndroid="rgba(0,0,0,0)"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "borderBottomColor": "#B2BEB5",
                  "borderBottomWidth": 0.5,
                  "height": 1,
                  "marginBottom": 20,
                  "marginTop": 20,
                  "width": "90%",
                }
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                  "width": "100%",
                },
              ]
            }
            virtual={undefined}
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#F7C04C",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 35,
                  "justifyContent": "center",
                  "marginTop": 15,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "width": "100%",
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {},
                  ]
                }
              >
                OK
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                  "width": "100%",
                },
              ]
            }
            virtual={undefined}
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#F7C04C",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 35,
                  "justifyContent": "center",
                  "marginTop": 15,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "width": "100%",
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {},
                  ]
                }
              >
                CANCEL
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
